module/VuFind/src/VuFind/View/Helper/Root/SortFacetList.php

```diff
@@ -67,9 +69,27 @@ public function __invoke($results, $field, $list, $searchRoute)
             $url = $urlHelper($searchRoute) . $results->getUrlQuery()
                 ->addFacet($field, $value['value'], ($value['operator'] ?? 'AND'))
                 ->getParams();
-            $facets[$url] = $value['displayText'];
+            $facets[$url] = $this->facetValueToString($value, $formatString);
         }
         $this->getSorter()->natsort($facets);
         return $facets;
     }
+
+    /**
+     * Convert a facet array to a string.
+     *
+     * @param array  $value  Facet value array
+     * @param string $format Format string
+     *
+     * @return string
+     */
+    protected function facetValueToString(array $value, string $format): string
```

Comment:
This mechanism is also used in the translations and might be useful in other cases. So I wonder if it is worth to create an Trait for this.

Reply:
This isn't exactly the same as what happens in the translations; while by convention we use the same `%%` tokens in translation strings and here, the translation system is actually doing a search-and-replace for arbitrary strings. In other words, the `%%` characters are already coded into the array of values sent to the translator. In this instance, we're turning an existing array into a set of tokens by adding the `%%` symbols around the array key names.

Is it possible there's some other instance more directly similar to this code that I'm forgetting about? If so, I'd be happy to refactor to a trait... but as far as I can recall, this is at least somewhat unique for the moment.

Reply:
Something similar also is done in LocalFile.php https://github.com/vufind-org/vufind/blob/12efd6a78a5765f826083c38a385575dd946f38b/module/VuFind/src/VuFind/Content/Covers/LocalFile.php#L120

But I also don't think that this justifies creating a trait for this.